etna/records/views/records.py

```diff
@@ -124,3 +147,34 @@ def record_detail_view(request, id):
 
     # Note: This page uses cookies to render GTM, please ensure to keep TemplateResponse or similar when changed.
     return TemplateResponse(request=request, template=template_name, context=context)
+
+
+def record_iiif_manifest_view(
+    request: HttpRequest, id: str
+) -> JsonResponse | HttpResponse:
+    """
+    Serve the IIIF manifest view to the client.
+
+    This proxies the manifest from CIIM (Rosetta API) to the client.
+
+    This view assumes that whatever is available at the API endpoint,
+    is safe to be served to the public.
+    """
+    try:
+        # This assumes that the ID passed in the request from the Internet
+        # is safe enough to pass it directly to the IIIF manifest API.
+        iiif_manifest = records_client.fetch_iiif_manifest(id=id)
+    except DoesNotExist:
+        raise Http404(id)
+    except Exception:
+        logger.warning(
+            "Unexpected error happened when trying to fetch the IIIF manifest for a record: requested_record_id=%s",
+            id,
+            exc_info=True,
+        )
+        # Return "503 Service Unavailable" if the IIIF manifest cannot be fetched
+        # for reasons other than it does not exist.
+        return HttpResponse(status=503)
+
+    # Proxy the IIIF manifest as-is from the CIIM (Rosetta) API to the client.
+    return JsonResponse(iiif_manifest.content)
```

Comment:
**Issue (blocking):** I think there's a big problem with using this proxy. The IIIF manifest contains links to the API that must be resolvable by the end user. I'm not sure how we want to proceed with that.

Reply:
Sorry Tomasz, I can't view the link you shared as it's behind Torchbox's slack. I've not seen the manifests etc so not sure what the real difference is apart from them being separate (for now)

Reply:
Sorry! It was a thread on our shared channel with Lee. I've got rid of the proxy view for now because it does not seem feasible to implement it now.